pkg/daemon/ceph/osd/daemon.go

```diff
@@ -202,9 +202,11 @@ func Provision(context *clusterd.Context, agent *OsdAgent, crushLocation, topolo
 		// Ideally, we would use the "ceph-volume inventory" command instead
 		// However, it suffers from some limitation such as exposing available partitions and LVs
 		// See: https://tracker.ceph.com/issues/43579
-		rawDevices, err = clusterd.DiscoverDevices(context.Executor)
-		if err != nil {
-			return errors.Wrap(err, "failed initial hardware discovery")
+		for i := range agent.devices {
+			rawDevices, err = clusterd.DiscoverDevicesWithFilter(context.Executor, agent.devices[i].IsFilter, agent.devices[i].Name)
+			if err != nil {
+				return errors.Wrap(err, "failed initial hardware discovery")
+			}
 		}
```

Comment:
```suggestion
		rawDevices, err = clusterd.DiscoverDevicesWithFilter(context.Executor, agent.devices[i].IsFilter, agent.devices[i].Name)
			if err != nil {
				return errors.Wrap(err, "failed initial hardware discovery")
			}
```

also is this loop required as ` DiscoverDevicesWithFilter` already loop through devices?

Reply:
Unfortunately yes, because **agent** is a struct that contains a slice of `osd.DesiredDevice{}`, which is created here https://github.com/rook/rook/blob/master/cmd/rook/ceph/osd.go#L211 what do you think?